exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/stat/ParquetMetaStatCollector.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.drill.common.types.Types;
+import org.apache.drill.exec.store.parquet.Metadata;
+import org.apache.drill.exec.store.parquet.ParquetGroupScan;
+import org.apache.parquet.column.statistics.BinaryStatistics;
+import org.apache.parquet.column.statistics.DoubleStatistics;
+import org.apache.parquet.column.statistics.FloatStatistics;
+import org.apache.parquet.column.statistics.IntStatistics;
+import org.apache.parquet.column.statistics.LongStatistics;
+import org.apache.parquet.column.statistics.Statistics;
+import org.apache.parquet.schema.OriginalType;
+import org.apache.parquet.schema.PrimitiveType;
+import org.joda.time.DateTimeConstants;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ParquetMetaStatCollector implements  ColumnStatCollector{
+  private  final Metadata.ParquetTableMetadataBase parquetTableMetadata;
+  private  final List<? extends Metadata.ColumnMetadata> columnMetadataList;
+  final Map<String, String> implicitColValues;
+
+  public ParquetMetaStatCollector(Metadata.ParquetTableMetadataBase parquetTableMetadata, List<? extends Metadata.ColumnMetadata> columnMetadataList, Map<String, String> implicitColValues) {
+    this.parquetTableMetadata = parquetTableMetadata;
+    this.columnMetadataList = columnMetadataList;
+    this.implicitColValues = implicitColValues;
+  }
+
+  @Override
+  public Map<SchemaPath, ColumnStatistics> collectColStat(Set<SchemaPath> fields) {
+    // map from column to ColumnMetadata
+    final Map<SchemaPath, Metadata.ColumnMetadata> columnMetadataMap = new HashMap<>();
+
+    // map from column name to column statistics.
+    final Map<SchemaPath, ColumnStatistics> statMap = new HashMap<>();
+
+    for (final Metadata.ColumnMetadata columnMetadata : columnMetadataList) {
+      SchemaPath schemaPath = SchemaPath.getCompoundPath(columnMetadata.getName());
+      columnMetadataMap.put(schemaPath, columnMetadata);
+    }
+
+    for (final SchemaPath schemaPath : fields) {
+      final PrimitiveType.PrimitiveTypeName primitiveType;
+      final OriginalType originalType;
+
+      final Metadata.ColumnMetadata columnMetadata = columnMetadataMap.get(schemaPath);
+
+      if (columnMetadata != null) {
+        final Object min = columnMetadata.getMinValue();
+        final Object max = columnMetadata.getMaxValue();
+        final Long numNull = columnMetadata.getNulls();
+
+        primitiveType = this.parquetTableMetadata.getPrimitiveType(columnMetadata.getName());
+        originalType = this.parquetTableMetadata.getOriginalType(columnMetadata.getName());
+        final Integer repetitionLevel = this.parquetTableMetadata.getRepetitionLevel(columnMetadata.getName());
+
+        statMap.put(schemaPath, getStat(min, max, numNull, primitiveType, originalType, repetitionLevel));
+      } else {
+        final String columnName = schemaPath.getRootSegment().getPath();
+        if (implicitColValues.containsKey(columnName)) {
```

Comment:
Ok, I was thinking that you may not even pass in the implicit columns to the expression materializer and let the PruneScanRule rule handle the dirN column filters by pushing down into the scan.. but since your new rule is applied later (during physical planning phase) it implies that those  were not removed.  So now if we want to leverage them for row-group pruning, it makes sense to create the materialized expression with them.